### ISSUE TITLE
cocos gen-libs does not work for android

### ISSUE THREAD

**Issue by crazyhappygame** [cocos2d/cocos2d-x#18790]
- cocos2d-x version: latest 
- developing environments
   - NDK version: 16b
   - Xcode version:
   - VS version:

Steps to Reproduce:
For latest version call:

`cocos gen-libs -p android -m release --ap android-15 --app-abi armeabi-v7a:x86`

Please see also PR with this failure
https://github.com/cocos2d/cocos2d-x/pull/18787

Error:
```
g:\cocos_latest\subrepos\repos>cocos gen-libs -p android -m release --ap android-15 --app-abi armeabi-v7a:x86
running: 'g:\cocos_latest\subrepos\external\cocos2d-x\tools/cocos2d-console/bin\cocos.bat compile -s g:\cocos_latest\subrepos\external\cocos2d-x\tests/cpp-empty-test -p android --mode release --app-abi armeabi-v7a:x86 --ap android-15'

Building mode: release
Using Android Studio project : g:\cocos_latest\subrepos\external\cocos2d-x\tests\cpp-empty-test\proj.android
Building apk...
Please input the absolute/relative path of '.keystore' file:
```

**Comment by crazyhappygame** [comment]
@drelaptop Could you check it?

**Comment by slackmoehrle** [comment]
@drelaptop This works for me if I don't build for __armeabi-v7a__ and __x86__ at the same time. If I build them separately it works, meaning something like this:
`cocos gen-libs -p android -m release --ap android-15 --app-abi armeabi-v7a`

**Comment by crazyhappygame** [comment]
The same error is for one architecture "--app-abi armeabi-v7a" and for two "--app-abi armeabi-v7a:x86"
`cocos gen-libs -p android -m release --ap android-15 --app-abi armeabi-v7a`

**Comment by crazyhappygame** [comment]
In PR there is only one architecture and it fails
https://github.com/cocos2d/cocos2d-x/pull/18787

https://travis-ci.org/cocos2d/cocos2d-x/jobs/370906160
https://travis-ci.org/cocos2d/cocos2d-x/jobs/370906161

**Comment by slackmoehrle** [comment]
```sh
(dubb @ dubbs-MBP ~/Cocos/cocos2d-x)  git:(v3) 3M 1A $ git remote -v
origin	git@github.com:cocos2d/cocos2d-x.git (fetch)
origin	git@github.com:cocos2d/cocos2d-x.git (push)
(dubb @ dubbs-MBP ~/Cocos/cocos2d-x)  git:(v3) 3M 1A $ git branch -v
  0v 18709f38d1 Update config.json
* v3 9275dda539 [behind 2] adjust android / linux build config for cocos CLI (#18780)
```

```sh
(dubb @ dubbs-MBP ~/Cocos/cocos2d-x)  git:(v3 2↓) 3M 1A $ cocos gen-libs -p android -m release --ap android-22 --app-abi armeabi-v7a
```

```sh
BUILD SUCCESSFUL in 10s
54 actionable tasks: 3 executed, 51 up-to-date
Move apk to /Users/dubb/Cocos/cocos2d-x/tests/cpp-empty-test/bin/debug/android
Build succeed.
```

```sh
(dubb @ dubbs-MBP ~/Cocos/cocos2d-x/tools/cocos2d-console/bin)  git:(upstream ⚡ detached@03d1f8d) $ cocos -v
Python 2.7.10
cocos2d-x-3.17
Cocos Console 2.3
```

**Comment by drelaptop** [comment]
@crazyhappygame for the discuss https://github.com/cocos2d/cocos2d-x/pull/18780 I know you are just wondering why need `.keystore` file for the release prebuilt libs.

I will check it, modify it or make a explain. the 3.17 round 2 test is going on.

**Comment by crazyhappygame** [comment]
@slackmoehrle you was tested with old cocos2d-x/tools/cocos2d-console. please check "git log".
git log: upstream ⚡ detached@03d1f8d
You are here: 03d1f8d

```
commit 03d1f8d291ed63439918fa4b180c7b9d698a2bfd (HEAD)
Author: minggo <xiaoming.zhang@chukong-inc.com>
Date:   Thu Jan 11 10:20:17 2018 +0800

    remove unneeded codes (#436)
```

You should be here: 1b69aa81a05d727cd15d5d41c94c0e7332bb622e 
```
commit 1b69aa81a05d727cd15d5d41c94c0e7332bb622e (HEAD -> v3, origin/v3, origin/HEAD)
Author: leda <li_cheng_zhe@163.com>
Date:   Tue Apr 24 09:36:37 2018 +0800

    fix gen-libs error on android, update version info for vs (#439)

    * fix gen-libs error, and update help info for vs

    * refactor ndk-mode -> mode
```

**Comment by crazyhappygame** [comment]
@drelaptop Yes. Please see https://github.com/cocos2d/cocos2d-x/pull/18787 I would like to create CI build for cocos gen-libs as well. In that way we will be sure that it really works and we will not introduce regressions.

**Comment by slackmoehrle** [comment]
@crazyhappygame I will test this with your steps above and yes, I am using cocos2d-console from within the Cocos2d-x root. I verified with `which cocos` before I even started since I have another cocos2d-x install for my games.

**Comment by v1993** [comment]
I tried setting `.keystore` file (note: it doesn't take my realtive path as valid, only absolute) and got anoter error:

    FAILURE: Build failed with an exception.
    
    * What went wrong:
    Execution failed for task ':CppEmptyTest:packageRelease'.
    > Failed to generate v1 signature
UPD: I got this problem with command `cocos gen-libs -p android -m release --ap android-16 --app-abi armeabi-v7a`.

**Comment by slackmoehrle** [comment]
@drelaptop after testing today, I too get asked about a `.keystore` file and I am not sure what to use. My understanding is that this is cached in `gradle.properties` but I don't see how it is needed for `cocos gen-libs`. This is a new issue. I wasn't getting this before @crazyhappygame has outlined the commits that caused this issue.

**Comment by drelaptop** [comment]
@slackmoehrle After reviewing the `cocos-console`, I found the `cocos gen-libs` call `cocos compile`, and then copy the generated libs. `cocos compile -m release` always check the `.keystore` path. 

Solutions occur to me, one is to delete the logic of check `.keystore` file, for release mode we can select to add the key infos into `gradle.properties`, not add it by cmd input. another is to add a option to `cocos compile`, for example `-no_key_input`, avoid the `.keystore` file.

**Comment by slackmoehrle** [comment]
-no_key_input

Seems like a better idea 

If I build debug, everything works:
```sh
$ cocos gen-libs -p android -m debug --ap android-22 --app-abi armeabi-v7a
```

**Comment by crazyhappygame** [comment]
@drelaptop Please let me know when I can test fixed `cocos gen-libs -p android -m release`

**Comment by drelaptop** [comment]
Correspond PR: https://github.com/cocos2d/cocos2d-console/pull/440

1. avoid to input sign infos when`gen-libs` by supporting generate no-signed apk
2. fix multi abi `--app-abi armeabi-v7a:x86` error, when `gen-libs`

**Comment by crazyhappygame** [comment]
@drelaptop I tested https://github.com/cocos2d/cocos2d-console/pull/440 and found two issues for:
`cocos gen-libs -p android -m release --ap android-15 --app-abi armeabi-v7a`
1. In command line there is APP_PLATFORM=android-14
Starting process 'command 'G:\cocos_latest\subrepos\3rdparty\android-ndk-r16b\ndk-build.cmd''. Working directory: G:\cocos_latest\subrepos\external\cocos2d-x\tests\cpp-empty-test\proj.android\app Command: G:\cocos_latest\subrepos\3rdparty\android-ndk-r16b\ndk-build.cmd NDK_PROJECT_PATH=null APP_BUILD_SCRIPT=G:\cocos_latest\subrepos\external\cocos2d-x\tests\cpp-empty-test\proj.android\app\jni\Android.mk NDK_APPLICATION_MK=G:\cocos_latest\subrepos\external\cocos2d-x\tests\cpp-empty-test\proj.android\app\jni\Application.mk APP_ABI=armeabi-v7a NDK_ALL_ABIS=armeabi-v7a NDK_DEBUG=0 APP_PLATFORM=android-14 NDK_OUT=G:/cocos_latest/subrepos/external/cocos2d-x/tests/cpp-empty-test/proj.android/app/build/intermediates/ndkBuild/release/obj NDK_LIBS_OUT=G:\cocos_latest\subrepos\external\cocos2d-x\tests\cpp-empty-test\proj.android\app\build\intermediates\ndkBuild\release\lib NDK_TOOLCHAIN_VERSION=clang -j12 NDK_DEBUG=0 G:/cocos_latest/subrepos/external/cocos2d-x/tests/cpp-empty-test/proj.android/app/build/intermediates/ndkBuild/release/obj/local/armeabi-v7a/libcpp_empty_test.so
Successfully started process 'command 'G:\cocos_latest\subrepos\3rdparty\android-ndk-r16b\ndk-build.cmd''
2. In output there is suspicious
file or directory 'G:\cocos_latest\subrepos\external\cocos2d-x\tests\cpp-empty-test\proj.android\app\libs', not found
file or directory 'G:\cocos_latest\subrepos\external\cocos2d-x\tests\cpp-empty-test\proj.android\app\libs', not found
file or directory 'G:\cocos_latest\subrepos\external\cocos2d-x\tests\cpp-empty-test\proj.android\app\libs', not found

**Comment by crazyhappygame** [comment]
@drelaptop Could you update cococos2d-console git submodule in cocos2d-x repo so I can test it in CI https://github.com/cocos2d/cocos2d-x/pull/18787

**Comment by drelaptop** [comment]
update the submodule here: https://github.com/cocos2d/cocos2d-x/pull/18808, that PR will be merged in recent 2 days.

**Comment by crazyhappygame** [comment]
When I try to use cocos compile (with latest changes) I have following error
```
$ cocos compile -m release -p android --android-studio --ap android-19 --app-abi armeabi-v7a:x86 --ndk-mode release
Python 2.7.9
Current available platforms : ['win32', 'metro']. 'android' is not available.
```

**Comment by drelaptop** [comment]
I think something wrong with you environments, when I test this command, outputs as:

![cocos-compile](https://user-images.githubusercontent.com/26329291/39857498-2f9ecc7a-5466-11e8-8e0c-ac49b5055031.png)

**Comment by crazyhappygame** [comment]
@drelaptop I am testing on windows (it seems like you  are on mac) and it seems like you do not have latest cocos2d-console ('there is check for .keystore'). I will check it in CI when everything will be merged.

**Comment by crazyhappygame** [comment]
I have also problem with using cocos gen-libs for android... Could you travis CI when we check that cocos gen-libs really works and we can reuse it for games? Something like: https://github.com/cocos2d/cocos2d-x/pull/18787

**Comment by drelaptop** [comment]
yep, I am testing on mac, I have latest cocos2d-console. The `cocos compile -p android -m release` check keystore file, but `cocos gen-libs -p android -m release` not. if you want compile a android release apk and without sign informations , you should add `--no-sign` parameter, it won't add key infos.

I can check #18787 after that submodule update PR merged. Let us discuss that later.

**Comment by drelaptop** [comment]
this issue have been fixed at cocos2d/cocos2d-console#440, and have been updated into cocos2d-x, so close this issue.

the next issue is how to find a proper way to use the generated libs. discuss here #18787
